docs/primitives/pubsub.md

```diff
@@ -69,6 +69,77 @@ the same message, the message will be delivered twice.
 
 </Callout>
 
+### Ordered Topics
+
+By default, topics are unordered, meaning that messages can be delivered in any order. This allows for better throughput
+on the topic as messages can be processed in parallel. However in some cases it's important that messages are delivered
+in the order they were published for a given entity, such as a user or shopping cart.
+
+To create an ordered topic, you need to configure the topic's `OrderingAttribute` to match the `pubsub-attr` tag on one
+of the top level fields of the event type. This field will ensure that messages delivered to the same subscriber will
+be delivered in the order of publishing for that specific field value. Messages with a different value on the ordering
+attribute can be delivered in any order.
+
+<Callout type="warning">
+
+By enabling ordered topics, you create a head-of-line blocking problem. This is because subsequent messages for the
+same ordering key will not be delivered until a message is successfully processed or dead-lettered. As such, it's
+strongly recommended to only use ordered topics when absolutely necessary and to ensure that your subscription configuration
+has a sensible retry policy which will dead-letter messages that fail to process within a timeframe that makes sense for
+your use case.
```

Comment:
```suggestion
With an ordered topic subequent messages for the same ordering key are not delivered until the earliest
message (for that ordering key) is successfully processed or dead-lettered. This can lead to a
[[head-of-line blocking](https://en.wikipedia.org/wiki/Head-of-line_blocking)](https://en.wikipedia.org/wiki/Head-of-line_blocking) problem,
and cause unexpected delays in message processing. Make sure you have sufficient logging and alerting in
place to handle processing issues, and make sure the retry policy on each subscription is configured
appropriate. In general, use ordered topics with care and consider whether the benefits outweigh these downsides.
```